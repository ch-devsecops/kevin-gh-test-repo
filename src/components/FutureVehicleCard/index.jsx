import React, { useContext } from 'react';
import {
  Box,
  Image,
  Label,
  Markdown,
  MediaItem,
  MarkdownHeading,
  useMediaQueries,
  useThemeContext,
} from '@honda-canada/design-system-react';
import { ModelFiltersContext } from '../ModelFiltersContext';
import { mapGTMCategory } from '../../utils/sitecoreFields';
import CTA from '../CTA';
import { HONDA_THEME_NAME } from '../../utils/constants';

const FutureVehicleCard = ({ fields, rendering }) => {
  const { isMobile } = useMediaQueries();
  const theme = useThemeContext();
  const modelFiltersContext = useContext(ModelFiltersContext);
  const { filterHasValues, filter } = modelFiltersContext || {};

  // This component is filterable. See ModelListFilters/reducer.js
  if (!fields || filterHasValues || filter.onlyNsx) return null;

  const gtmCategory = fields?.gtmCategory?.value;
  const { anchorId, title, bodyText, ctaLink, ctaType, ctaIcon, bannerLabel, gtmTitle, gtmInteractionType } = fields;
  const image = isMobile ? fields?.mobileImage : fields?.desktopImage;
  return (
    <Box
      data-gtm-category={mapGTMCategory(gtmCategory)}
      data-gtm-component-type={rendering?.componentName}
      pb={['m', 'l']}
      mb="m"
      id={anchorId?.value}
      backgroundColor="white"
    >
      <Box>
        {bannerLabel?.value && (
          <Box
            width="100%"
            height={['28px', '38px']}
            backgroundColor="blue"
            zIndex="100"
            display="flex"
            justifyContent="center"
            flexDirection="column"
            textAlign="center"
          >
            <Label color="white" fontFamily="bold" fontSize={['16px', '18px']}>
              {bannerLabel?.value}
            </Label>
          </Box>
        )}
        <MediaItem image={<Image {...image?.value} />} containerSize="100%" />
      </Box>

      <Box mt={['l', 'xl']}>
        <Box my="m">
          {title?.value && (
            <MarkdownHeading
              headingOverride="h5"
              textAlign="center"
              mb={theme.name === HONDA_THEME_NAME ? 'xs' : ['s', 'm']}
              mx="m"
              fontFamily="bold"
            >
              {title?.value}
            </MarkdownHeading>
          )}
          <Markdown mx={['l', 0]} textAlign="center" mb={0} letterSpacing="0.5px">
            {bodyText?.value}
          </Markdown>
        </Box>
        <Box textAlign="center">
          {ctaLink?.value?.href && (
            <CTA
              linkField={ctaLink}
              typeField={ctaType}
              iconField={ctaIcon}
              gtmTags={{
                'data-gtm-interaction-type': gtmInteractionType?.value || 'cta: click',
                'data-gtm-title': gtmTitle?.value,
                'aria-label': ctaLink?.value?.title,
              }}
            />
          )}
        </Box>
      </Box>
    </Box>
  );
};
export default FutureVehicleCard;
