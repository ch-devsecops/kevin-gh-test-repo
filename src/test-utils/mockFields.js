/* eslint-disable max-len */
/* eslint-disable max-classes-per-file */

// FIXME: jsdoc popup not working on test files
/** Class representing a mock data. */
class MockData {
  /**
   * Add a field.
   * @param {string} key - key of the field obj: i.e. 'title'
   * @param {Object} value - value of the field obj: i.e. { value: 'some text' }.
   */
  add(key, value) {
    this[key] = value;
  }
}

/**
 * Class representing mock fields object.
 * @extends MockData
 *
 * available method: add()
 *
 * default fields: [anchorId, title, bodyText, image, ctaLink, ctaType, ctaIcon, gtmCategory, gtmTitle, contentTitle, subtitle]
 *
 * note: we can add more commonly used fields in the class declaration if needed
 */
class MockFields extends MockData {
  constructor() {
    super();

    this.title = { value: 'Title' };
    this.bodyText = { value: 'BodyText' };
    this.image = {
      value: { src: 'mock.png', alt: 'mock alt' },
    };
    this.ctaLink = {
      value: {
        href: 'www.source.ca',
        title: 'cta aria label',
        text: 'CTA Text',
      },
    };
    this.ctaType = {
      value: 'Tertiary',
    };
    this.ctaIcon = {
      fields: {
        value: {
          value: 'arrowRight',
        },
      },
    };
    this.gtmCategory = {
      fields: {
        value: {
          value: 'Others',
        },
      },
    };
    this.gtmTitle = {
      value: 'gtm title',
    };
    this.contentTitle = {
      value: 'Content Title',
    };
    this.subtitle = {
      value: 'Content Subtitle',
    };
    this.seoH1 = {
      value: '',
    };
    this.anchorId = {
      value: 'anchor-id',
    };
    this.ctaLabel = {
      value: 'CTA Label',
    };
  }
}

/**
 * Class representing a mock rendering object
 * @extends MockData
 *
 * available method: add()
 */
class MockRendering extends MockData {
  /**
   * Create a mock rendering object
   * @param {string} componentName - adds componentName
   */
  constructor(componentName) {
    super();

    this.componentName = componentName;
  }
}

export { MockFields, MockRendering };
