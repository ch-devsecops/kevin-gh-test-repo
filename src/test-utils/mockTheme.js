/**
 * Mock styled-components theme, if you want to test how your component
 * implements theme styles. Import as-is, or spread in any
 * additional values you need.
 *
 * Example usage:

  it('should apply the bgColor param', () => {
    const { colors } = MockTheme;
    const { container } = render(
      <MyComponent
        fields={mockFields}
        params={{ bgColor: 'Gray' }}
      />,
    );
    const wrapper = container.querySelector('section');

    expect(wrapper).toHaveStyleRule('background-color', colors.grey[5]);
  });
 */

import { HONDA_THEME_NAME } from '../utils/constants';

const mockTheme = {
  letterSpacings: {
    sm: '0.5px',
    md: '1px',
  },
  lineHeights: {
    sm: '22px',
    md: '26px',
  },
  colors: {
    white: '#fff',
    transparent: 'transparent',
    grey: ['#707070', '#C7C7C7', '#D5D5D5', '#EAEAEA', '#F8F8F8', '#F5F5F5'],
    black: '#2C2C2C',
    red: '#DA2C2E',
    redShades: ['rgba(200, 16, 46, 0.1)'],
    aodaRed: '#D92629',
    lightRed: '#EC9597',
    darkRed: '#A21C1D',
    blue: '#334C96',
    darkBlue: '#002D62',
    lightBlue: '#00447826',
    yellow: ['#F9CA52', '#EDAB00', '#8D763F'],
    fadedBlue: '#ADB7D5',
    primary: '#DA2C2E',
    primaryAoda: '#D92629',
    primaryDark: '#A21C1D',
    primaryLight: '#EC9597',
    secondary: '#2C2C2C',
    secondaryLight: '#C7C7C7',
    tertiary: '#002D62',
    tertiaryLight: '#334C96',
    tertiaryDisabled: '#ADB7D5',
    danger: '#DA2C2E',
    warning: '#2C2C2C',
    warningLight: '#EAEAEA',
    success: '#334C96',
    successLight: '#d9e0e7',
    disabled: '#C7C7C7',
    typographyDefault: '#2C2C2C',
  },
  space: [0, 4, 8, 12, 16, 24, 32, 48, 64, 96, 128, 144, 192, 256],
  borders: [0, '1px solid', '2px solid', '3px solid'],
  breakpoints: ['768px', '992px', '1200px'],
  fontWeights: {
    regular: 'normal',
    bold: 700,
  },
  fontSizes: {
    sm: '12px',
  },
  shadows: {
    smooth: {
      md: '0.1px 0.3px 0.3px rgba(0, 0, 0, 0.02),0.3px 0.7px 0.9px rgba(0, 0, 0, 0.028),0.7px 1.4px 1.8px rgba(0, 0, 0, 0.035),1.5px 2.9px 3.7px rgba(0, 0, 0, 0.047),4px 8px 10px rgba(0, 0, 0, 0.07)',
    },
    box: {
      subNav: '0px 2px 4px rgba(0, 0, 0, 0.1)',
    },
  },
  zIndices: {
    accordionItemTitle: 200,
    modal: 13001,
    sectionNav: 13000,
  },
  transitionDuration: {
    t1: '1s',
  },
  transitionTimingFunction: {
    default: 'ease-in-out',
    out: 'ease-out',
    in: 'ease-in',
  },
  name: HONDA_THEME_NAME,
  fonts: {
    default: 'Helvetica',
    bold: 'Cooper Black',
    heading: 'Roboto',
  },
  header: {
    desktop: {},
    mobile: {},
  },
  icons: {},
  logIconWarnings: false,
};

export default mockTheme;
