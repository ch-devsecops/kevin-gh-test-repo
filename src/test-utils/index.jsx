import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { BrowserRouter } from 'react-router-dom';
import { OptionalVideoModalContextProvider } from '@honda-canada/design-system-react';
import UserLocationProvider from '@honda-canada/user-location';
import { configureAxe } from 'jest-axe';
import ModelExplorationContextProvider from '../components/ModelExplorationContext';
import ModelFiltersContextProvider from '../components/ModelFiltersContext';
import { MockFields, MockRendering } from './mockFields';
import MockTheme from './mockTheme';
import 'intersection-observer';
import { ACURA_THEME_NAME } from '../utils/constants';

const generateJssWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ACURA_THEME_NAME },
          ...context?.sitecoreContext,
        }}
        componentFactory={{}}
      >
        <BrowserRouter>
          <UserLocationProvider>
            <ThemeProvider theme={context?.theme || MockTheme}>
              <OptionalVideoModalContextProvider>
                <ModelExplorationContextProvider>
                  <ModelFiltersContextProvider>{children}</ModelFiltersContextProvider>
                </ModelExplorationContextProvider>
              </OptionalVideoModalContextProvider>
            </ThemeProvider>
          </UserLocationProvider>
        </BrowserRouter>
      </SitecoreContext>
    );
  };
/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in an typical JSS app's context providers.
 * Note that you can include a `context` object to set or override values
 * for SitecoreContext and ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.context] - default values for SitecoreContext, ThemeProvider
 * @param {string} [options.context.sitecoreContext] - value for SitecoreContext
 * @param {string} [options.context.theme] - value for ThemeProvider
 * @example
 * const { getByText } = jssRender(
 *  <MyComponent />, { context: { sitecoreContext: { language: 'fr' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const jssRender = (component, options = {}) => {
  const { context } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.context;

  return render(component, {
    wrapper: generateJssWrapper(context),
    ...options,
  });
};

const generateWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <BrowserRouter>
        <ThemeProvider theme={context?.theme || MockTheme}>
          <OptionalVideoModalContextProvider>{children}</OptionalVideoModalContextProvider>
        </ThemeProvider>
      </BrowserRouter>
    );
  };

/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in typical design-system context providers.
 * Note that you can include a `theme` object to set or override values
 * for ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.theme] - default value for ThemeProvider
 * @example
 * const { getByText } = render(
 *  <MyComponent />, { theme: { space: { wrapperGap: '15px' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const customRender = (component, options = {}) => {
  const { theme } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.theme;

  return render(component, {
    wrapper: generateWrapper(theme),
    ...options,
  });
};

// disable accessibility checks for errors that are usually trigged by content mistakes
const axe = configureAxe({
  rules: {
    'duplicate-id': { enabled: false },
    'heading-order': { enabled: false },
  },
});

export * from 'jest-axe';
export { axe };
export * from '@testing-library/react';
export { customRender as render, jssRender };
export { MockFields, MockRendering, MockTheme };
