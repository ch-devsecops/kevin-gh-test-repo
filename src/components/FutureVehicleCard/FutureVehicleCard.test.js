import React from 'react';
import { jssRender } from 'test-utils';
import FutureVehicleCard from './index';

const routeData = require('../../../sample-mock-data/routes/future-vehicle-card/en.json');

const props = routeData.placeholders['jss-main'][0];

describe('<FutureVehicleCard>', () => {
  it('should render without errors', () => {
    const { container } = jssRender(<FutureVehicleCard {...props} />);

    expect(container).toMatchSnapshot();
  });
});
