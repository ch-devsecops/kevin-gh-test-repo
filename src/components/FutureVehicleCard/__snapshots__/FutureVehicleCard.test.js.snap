// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FutureVehicleCard> should render without errors 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 frGmdx"
    data-gtm-category="Others"
  >
    <div
      class="Box-sc-1bsf784-0 iPRsIc"
    >
      <div
        class="Box-sc-1bsf784-0 iCPSZa"
        display="flex"
        height="28px,38px"
        width="100%"
      >
        <label
          class="Label-sc-115wxjh-0 bjMVgg"
          color="white"
          font-family="bold"
          font-size="16px,18px"
        >
          Future Vehicle #1
        </label>
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 dYxbmH"
      >
        <img
          class="sc-1k44dfv-0 jLTFZk"
          draggable="true"
          loading="lazy"
        />
      </div>
    </div>
    <div
      class="Box-sc-1bsf784-0 hLoUhz"
    >
      <div
        class="Box-sc-1bsf784-0 cPxeOK"
      >
        <div
          class="sc-1k44dfv-0 bfxkBk"
          font-family="bold"
          mb="xs"
          mx="m"
          textalign="center"
        >
          

          <h5
            class="sc-1k44dfv-0 gXWcpP"
          >
            The New 2023 Integra
          </h5>
        </div>
        <div
          class="sc-1k44dfv-0 gCCWVx"
          letter-spacing="0.5px"
          mb="0"
          mx="l,0"
          textalign="center"
        >
          

          <p>
            The all-new Integra ushers in a new generation of Precision Crafted PerformanceÂ®. Arriving spring.
          </p>
        </div>
      </div>
      <div
        class="Box-sc-1bsf784-0 bvhfjY"
      >
        <span>
          <a
            class="Link-sc-16wd4a0-0 eYsnbh"
            data-gtm-interaction-type="cta: gtm interaction type"
            data-gtm-title="gtm-title"
            href="https://www.hondacanada.ca"
            target=""
          >
            CTA Link 1
          </a>
        </span>
      </div>
    </div>
  </div>
</div>
`;
